### PR TITLE
feat(typescript_indexer): don't emit duplicate nodes for namespaces

### PR DESCRIPTION
**Pull request by nbeloglazov**

### CODE REVIEW

File: kythe/typescript/testdata/namespace.ts [hunk 1]
```diff
@@ -17,3 +17,13 @@ namespace N {
...
 //- @obj ref NObj
 N.obj;
+
+// Repeat same namespace to make
+// sure we don't output duplicate
+// facts.
+//- SecondNdef defines NValue
+//- SecondNdef.loc/start @^"namespace"
+namespace N {
+	export const boo = 1;
```
Comment (evmar) [inline kythe/typescript/testdata/namespace.ts]: Fix whitespace here.

File: kythe/typescript/testdata/namespace.ts [hunk 2]
```diff
@@ -17,3 +17,13 @@ namespace N {
...
+  //- SecondNdef.loc/end @$"}"
+}
```
Comment (evmar) [inline kythe/typescript/testdata/namespace.ts]: What about something like:

```
namespace N.z {
}
```

Do we emit `N` again here?

File: kythe/typescript/testdata/namespace.ts (hunk 1 above)
Comment (nbeloglazov) [reply kythe/typescript/testdata/namespace.ts]: Done.

File: kythe/typescript/testdata/namespace.ts (hunk 2 above)
Comment (nbeloglazov) [reply kythe/typescript/testdata/namespace.ts]: Yes. We emit `defines` edge but not `defines/binding`. Added test for N.Z.

**Pull request merged**